spec/ios_get_build_number_spec.rb

```diff
@@ -0,0 +1,66 @@
+require 'spec_helper'
+
+describe Fastlane::Actions::IosGetBuildNumberAction do
+  describe 'getting the build number from the provided .xcconfig file' do
+    it 'parses an xcconfig file with keys without spacing and returns the correct build number' do
+      xcconfig_mock_content = <<~CONTENT
+        // a comment
+        VERSION_SHORT=6
+        VERSION_LONG=6.30.0
+        BUILD_NUMBER=1940
+      CONTENT
+
+      expect_build_number(xcconfig_mock_content: xcconfig_mock_content, expected_build_number: '1940')
+    end
+
+    it 'parses an xcconfig file with keys with spaces and returns a nil build number' do
+      xcconfig_mock_content = <<~CONTENT
+        VERSION_SHORT = 6
+        VERSION_LONG = 6.30.1
+        BUILD_NUMBER = 1940
+      CONTENT
+
+      expect_build_number(xcconfig_mock_content: xcconfig_mock_content, expected_build_number: nil)
+    end
```

Comment:
I was surprised by this because `key = value` is a valid `xcconfig` syntax. Looking at the pre-existing, underlying implementation, I can see why this is required given that we manually parse the `xcconfig` files

https://github.com/wordpress-mobile/release-toolkit/blob/e4b28a48dc760fe25be1d31d959de8c0b1726821/lib/fastlane/plugin/wpmreleasetoolkit/helper/ios/ios_version_helper.rb#L286-L302

We should be able to replace that bespoke logic with the API provided by the `xcodeproj` gem like [we do in Jetpack and WordPress iOS](https://github.com/wordpress-mobile/WordPress-iOS/blob/00a91b905cd1614cb272224114464fda6292aa77/fastlane/lanes/build.rb#L320-L321):


```ruby
version_config_path = File.join(PROJECT_ROOT_FOLDER, 'config', 'Version.internal.xcconfig')
versions = Xcodeproj::Config.new(File.new(version_config_path)).to_hash
```

Reply:
It looks like @iangmaia already started working on using `xcodeproj` instead of the manual parsing: https://github.com/wordpress-mobile/release-toolkit/pull/451

I decided to implement this as-is and update it as needed after Ian's PRs are fully merged into the project.